pkg/storage/odf.go

```diff
@@ -0,0 +1,279 @@
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, storageSystemObj)
+
+	if err != nil {
+		glog.V(100).Infof("failed to find storage SystemODF object %s in namespace %s",
+			builder.Definition.Name, builder.Definition.Namespace)
+
+		return nil, err
+	}
+
+	return storageSystemObj, nil
+}
+
+// Exists checks whether the given storage SystemODF exists.
+func (builder *SystemODFBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if storage SystemODF %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Create makes a storage SystemODF in the cluster and stores the created object in struct.
+func (builder *SystemODFBuilder) Create() (*SystemODFBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the storage SystemODF %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace,
+	)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Delete removes storage SystemODF object from a cluster.
+func (builder *SystemODFBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the storage SystemODF object %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		glog.V(100).Infof("storage SystemODF %s in namespace %s cannot be deleted"+
+			" because it does not exist",
+			builder.Definition.Name, builder.Definition.Namespace)
+
+		builder.Object = nil
+
+		return nil
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		return fmt.Errorf("can not delete storage SystemODF: %w", err)
+	}
+
+	builder.Object = nil
+
+	return nil
+}
+
+// WithStorageClusterSpec sets the storage SystemODF with storageCluster spec values.
+func (builder *SystemODFBuilder) WithStorageClusterSpec(
+	kind odfoperatorv1alpha1.StorageKind, name, nsname string) *SystemODFBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Setting storage SystemODF %s in namespace %s with storageCluster spec; \n"+
+			"kind: %v, name: %s, namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace, kind, name, nsname)
+
+	if kind == "" {
+		glog.V(100).Infof("The kind of the storageCluster spec is empty")
+
+		builder.errorMsg = "storageCluster spec 'kind' cannot be empty"
+
+		return builder
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the storageCluster spec is empty")
+
+		builder.errorMsg = "storageCluster spec 'name' cannot be empty"
```

Comment:
Please change to SystemODF  all logs/comments below

Reply:
this is not storageODF name, this is storageCluster name